src/main/java/core/basesyntax/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import java.util.Map;
+
+public class FruitDaoImpl implements FruitDao {
+
+    @Override
+    public Map<String, Integer> getStorage() {
+        return Storage.fruitsStorage;
+    }
+
+    @Override
+    public Integer get(String key) {
+        return Storage.fruitsStorage.get(key);
+    }
+
+    @Override
+    public void add(String name, Integer quantity) {
+        Storage.fruitsStorage.put(name, quantity);
```

Comment:
When adding a quantity of fruit to the storage, you should consider the case where the fruit already exists in the storage. In such a case, you should add the new quantity to the existing one instead of overwriting it.